pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -1035,6 +1044,10 @@ func (cr *PerconaXtraDBCluster) HAProxyEnabled() bool {
 	return cr.Spec.HAProxy != nil && cr.Spec.HAProxy.Enabled
 }
 
+func (cr *PerconaXtraDBCluster) HAProxyReplicasServiceEnabled() bool {
+	return cr.Spec.HAProxy.ReplicasServiceEnabled == nil || *cr.Spec.HAProxy.ReplicasServiceEnabled
```

Comment:
it's always not nil, because we check this value in CheckNSetDefaults

Reply:
just what to clarify: check for `nil` for pointer before dereference it to avoid panic in production is bad thing? (I think it costs ~2 CPU ticks)